packages/eslint-plugin/tests/rules/no-use-before-define.test.ts

```diff
@@ -219,7 +219,29 @@ type Foo = string | number;
       `,
       options: [{ typedefs: false }],
     },
+    // https://github.com/typescript-eslint/typescript-eslint/issues/2572
+    {
+      code: `
+interface Bar {
+  type: typeof Foo.FOO;
+}
+
+class Foo {
+  public static readonly FOO = '';
+}
+      `,
+      options: [{ ignoreTypeReferences: true }],
+    },
+    {
+      code: `
+const foo = 2;
 
+interface Bar {
+  type: typeof foo;
+}
+      `,
+      options: [{ ignoreTypeReferences: true }],
+    },
     // https://github.com/bradzacher/eslint-plugin-typescript/issues/141
     {
       code: `
```

Comment:
It would be good to add a few negative tests as well:
Use the same code samples, but with `ignoreTypeReferences: false`.

Reply:
Added!